app/modules/wallet/addresses/addresses.controller.ts

```diff
@@ -0,0 +1,56 @@
+import * as restify from 'restify';
+import { InternalServerError, NotFoundError } from 'restify-errors';
+import IController from '../../../interfaces/utils/IController';
+import IRequest from '../../../interfaces/utils/IServer';
+import WalletAddress from '../../../models/wallet_address.model';
+import { IWallet } from '../../../interfaces/models';
+
+export default class WalletsController implements IController {
+  public async put(req: IRequest, res: restify.Response, next: restify.Next) {
+    try {
+      const walletAddress = await WalletAddress.create({
+        chain: req.body.chain,
+        address: req.body.address,
+        name: req.body.name,
+        wallet_id: req.params.wallet
+      });
+
+      return res.send({
+        walletAddress
+      });
+    } catch (e) {
+      console.log(e);
+      throw new InternalServerError(e);
+    }
+    return next();
+  }
+
+  public async get(req: IRequest, res: restify.Response, next: restify.Next) {
+    try {
+      const walletAddresses = await WalletAddress.findAll({
+        where: { wallet_id: req.params.wallet }
+      });
+      if (!walletAddresses || walletAddresses.length === 0) {
+        return res.send(new restify.NotFoundError('Addresses not found for given wallet id'));
+      }
+      return res.send(walletAddresses);
+    } catch (e) {
+      console.log(e);
+      throw new InternalServerError(e);
+    }
+    return next();
+  }
+
+  public async delete(req: IRequest, res: restify.Response, next: restify.Next) {
+    try {
+      const walletAddresses = await WalletAddress.destroy({
+        where: { wallet_id: req.params.wallet, id: req.params.address }
+      });
+      return res.send({ message: 'Deleted Address' });
```

Comment:
We'll return `{success: true, message: '<message here>'}` throughout the API whenever something is successful.